### PR TITLE
256/progress bar

### LINKED ISSUE #256: Minimizable Progress Bar Implementation
Create a reusable progress bar component to be used for the File Upload and Processed Data Pages. See File Upload page design for design.

- [ ]  Add runtime Assertions

- [ ]  Create Jest tests

- [ ]  Log errors using rollbar

### PR DESCRIPTION
**Pull request by cameronbeattie306**
Now I am looking to merge.

Resolves #256

### CODE REVIEW

File: frontend/src/components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+        if (popup == null || overLay == null || maxButton == null) {
+            rollbar.error("Help popup element does not exist in document");
+        } else {
+            popup.hidden = true;
+            overLay.hidden = true;
+            maxButton.style.display = "block";
+        }
+    };
+
```
Comment (ralphgregorio) [inline frontend/src/components/ProgressBar/ProgressBar.tsx]: To make the progress bar reusable for all components, I think its better to have a prop called isVisible and this should control if the Progress bar should return or return null.

So handling the minimization of the progress bar would be handled outside this component.

File: frontend/src/components/ProgressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                            className={styles.minimizeButton}
+                            variant="contained"
+                            onClick={closePopup}
+                        >
+                            Minimize
+                        </Button>
+                    </div>
+
+                    <div className={styles.body}>
+                        Your task is being processed. Please wait...
```
Comment (ralphgregorio) [inline frontend/src/components/ProgressBar/ProgressBar.tsx]: I think we should also have a prop containing text, this way we can change what text is getting passed while its loading.

**Review by ralphgregorio — COMMENTED** [review]
Looking good! Just comments on generalization on the progress bar. And I think I will reuse this in my FileUpload Page.

**Comment by cameronbeattie306** [comment]
Thanks for the feedback! I'll add the prop for text. 

I'm not sure about the visibility though. I implemented closePopup and openPopup so the user can minimize the progress bar (and re open it if they choose). If you pass in a value less than 0 or greater than 100 the progress bar returns null.

I can definitely add that prop if you think it's better.

File: frontend/src/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (cameronbeattie306) [reply frontend/src/components/ProgressBar/ProgressBar.tsx]: The open/close popup functions are for handling the minimization requirement of the progress bar. I added the isVisible prop, so it returns null if that is false, or if the percentage < 0 or percentage > 100.

File: frontend/src/pages/ProcessDataPage/ProcessedDataPage.tsx [hunk 3]
```diff
@@ -73,7 +76,11 @@ const ProcessedDataPage = function () {
         if (currentFile) {
             const { id, watch } = currentFile;
             const lowerCaseWatch = watch.toLowerCase();
+
+            // TODO: Do a better job of guestimating percentages...
+            setPercentage(20);
             handlePredict(id, selectedModel, lowerCaseWatch);
```
Comment (ralphgregorio) [inline frontend/src/pages/ProcessDataPage/ProcessedDataPage.tsx]: Make sure this needs to be await to make sure this line finishes before the percentage finishes
```suggestion
            await handlePredict(id, selectedModel, lowerCaseWatch);
```

**Review by ralphgregorio — COMMENTED** [review]
> Thanks for the feedback! I'll add the prop for text.
> 
> I'm not sure about the visibility though. I implemented closePopup and openPopup so the user can minimize the progress bar (and re open it if they choose). If you pass in a value less than 0 or greater than 100 the progress bar returns null.
> 
> I can definitely add that prop if you think it's better.

Sure thinking about it, I think glenn mentioned we need to have them all minimizable so I think its good to keep it for now.

So as far as implementation to me it looks good just one minor issue on how you are setting percentages in ProcessPage

File: frontend/src/pages/ProcessDataPage/ProcessedDataPage.tsx (hunk 3 above)
Comment (cameronbeattie306) [reply frontend/src/pages/ProcessDataPage/ProcessedDataPage.tsx]: That makes sense. Thanks!

**Review by ralphgregorio — APPROVED** [review]
Looks great!

**Pull request merged**